lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -227,74 +204,82 @@ Point TruncatedDistribution::getRealization() const
 {
-  if (point.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << getDimension() << ", here dimension=" << point.getDimension();
+
+  for (UnsignedInteger k = 0; k < getDimension(); ++ k)
+  {
+    const Scalar xK = point[k];
+    if (xK <= bounds_.getLowerBound()[k]) return 1.0;
+    if (xK >= bounds_.getUpperBound()[k]) return 0.0;
+  }
 
-  const Scalar x = point[0];
-  if (x <= lowerBound_) return 1.0;
-  if (x > upperBound_) return 0.0;
   // If tail=true, don't call distribution_.computeCDF with tail=true in the next line!
   return normalizationFactor_ * (cdfUpperBound_ - distribution_.computeCDF(point));
 }
 
 /* Get the PDFGradient of the distribution */
 Point TruncatedDistribution::computePDFGradient(const Point & point) const
 {
-  if (point.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
 
   const Scalar x = point[0];
-  if ((x <= lowerBound_) || (x > upperBound_)) return Point(distribution_.getParametersCollection()[0].getDimension() + finiteLowerBound_ + finiteUpperBound_);
+  if ((x <= bounds_.getLowerBound()[0]) || (x > bounds_.getUpperBound()[0])) return Point(distribution_.getParametersCollection()[0].getDimension() + bounds_.getFiniteLowerBound()[0] + bounds_.getFiniteUpperBound()[0]);
   const Point pdfGradientX(distribution_.computePDFGradient(point));
-  const Point cdfGradientLowerBound(finiteLowerBound_ ? distribution_.computeCDFGradient(Point(1, lowerBound_)) : Point(distribution_.getParametersCollection()[0].getDimension()));
-  const Point cdfGradientUpperBound(finiteUpperBound_ ? distribution_.computeCDFGradient(Point(1, upperBound_)) : Point(distribution_.getParametersCollection()[0].getDimension()));
+  const Point cdfGradientLowerBound(bounds_.getFiniteLowerBound()[0] ? distribution_.computeCDFGradient(bounds_.getLowerBound()) : Point(distribution_.getParametersCollection()[0].getDimension()));
+  const Point cdfGradientUpperBound(bounds_.getFiniteUpperBound()[0] ? distribution_.computeCDFGradient(bounds_.getUpperBound()) : Point(distribution_.getParametersCollection()[0].getDimension()));
   const Scalar pdfPoint = distribution_.computePDF(point);
   Point pdfGradient(normalizationFactor_ * pdfGradientX - pdfPoint * normalizationFactor_ * normalizationFactor_ * (cdfGradientUpperBound - cdfGradientLowerBound));
   // If the lower bound is finite, add a component to the gradient
-  if (finiteLowerBound_)
+  if (bounds_.getFiniteLowerBound()[0])
```

Comment:
Wrong. All the finite lower bounds should be taken into account, not only the first one. The PDF values at the truncation interval vertices of the original distribution are needed here, see the comment for the constructor.